### PR TITLE
mirror-nixos-branch: omit minimal ISOs from small channels

### PR DESCRIPTION
**Pull request by emilazy**
See https://github.com/NixOS/nixpkgs/pull/338433.

I’m not very good at Perl and I don’t know how to test these scripts, so I’d appreciate some scrutiny here.

### CODE REVIEW

File: mirror-nixos-branch.pl [hunk 1]
```diff
@@ -214,8 +214,12 @@ sub fetch {
         downloadFile("nixpkgs.tarball", "packages.json.br", "json-br");
         downloadFile("nixos.options", "options.json.br", "json-br");
 
-        downloadFile("nixos.iso_minimal.aarch64-linux");
-        downloadFile("nixos.iso_minimal.x86_64-linux");
+        # Minimal installer ISOs were dropped from the small channel
+        if ($channelName !~ /-small/ ||
+            $channelName =~ /nixos-2([0123]|4\.05)-small/) {
```
Comment (vcunat) [inline mirror-nixos-branch.pl:219]:
```suggestion
            $channelName =~ /nixos-2([0123]\...|4\.05)-small/) {
```

**Review by vcunat — CHANGES_REQUESTED** [review]
I believe this is needed to trigger e.g. on `nixos-23.11-small`.  Otherwise it looks good to me.

File: mirror-nixos-branch.pl (hunk 1 above)
Comment (emilazy) [reply mirror-nixos-branch.pl]: Ah, of course; I only added the `-small` after. Fixed.

**Review by vcunat — APPROVED** [review]

**Comment by emilazy** [comment]
@vcunat https://github.com/NixOS/nixpkgs/pull/338433 has three 👍s now; do you think that’s adequate approval or would you like to see more response here before merge? (I linked it in the Matrix contributors room a few days ago to no response other than those; not sure how I could get more reactions from people who would particularly care.)

**Pull request merged**